tests/metrics/density/docker_memory_usage.sh.in

```diff
@@ -22,43 +22,89 @@
 #  This test launches certain number of containers and puts them into
 #  sleep mode for a certain period of time. Then the test checks the
 #  amount of memory used by each container
-#  This test uses psstop tool to get the memory used.
+#  This test uses smem tool to get the memory used.
 
 set -e
 
-[ $# -ne 2 ] && ( echo >&2 "Usage: $0 <times to run> <wait time>"; exit 1 )
+[ $# -ne 2 ] && ( echo >&2 "Usage: $0 <cc number> <wait time>"; exit 1 )
 
 SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
 source "${SCRIPT_PATH}/../../lib/test-common.bash"
 
 CMD='sh'
-IMAGE='ubuntu'
-TIMES="$1"
+IMAGE='busybox'
+CC_NUMBER="$1"
 WAIT_TIME="$2"
-TEST_NAME="Qemu Memory Usage"
+TEST_NAME="CC Memory Usage"
 TEST_ARGS="rootfs=${IMAGE} units=kb"
 TEST_RESULT_FILE=$(echo "${RESULT_DIR}/${TEST_NAME}" | sed 's| |-|g')
 SMEM_BIN=$(command -v smem || true)
 QEMU_BIN="@QEMU_PATH@"
+PROXY_BIN="@libexecdir@/cc-proxy"
+SHIM_BIN="@libexecdir@/cc-shim"
 
+# This function measures the PSS average
+# memory of a process.
+function get_pss_memory(){
+	ps="$1"
+	mem_amount=0
+	count=0
+	avg=0
+
+	data=$("$SMEM_BIN" --no-header -P "^$ps" -c "pss")
+	for i in $data;do
+		if (( $i > 0 ));then
+			mem_amount=$(( $i + $mem_amount ))
+			count=$(( $count + 1 ))
+		fi
+	done
+
+	if (( $count > 0 ));then
+		avg=$(echo "scale=2; $mem_amount / $count" | bc -l)
+	fi
+
+	echo "$avg"
+}
+
+# It calculates the memory footprint
+# of a CC.
 function get_docker_memory_usage(){
+	qemu_mem=0
+	cc_shim_mem=0
+	proxy_mem=0
+	cc_proxy_mem=0
+
 	containers=()
-	for i in $(seq 1 "$TIMES"); do
+
+	for i in $(seq 1 "$CC_NUMBER"); do
 		containers+=($(random_name))
 		${DOCKER_EXE} run --name ${containers[-1]} -tid $IMAGE $CMD
 	done
+
+	# This time will determine if the data
+	# will be taken w/o KSM working.
 	sleep "$WAIT_TIME"
-	for i in $("$SMEM_BIN" --no-header -P "^$QEMU_BIN" | awk '{print $6}'); do
-		test_data=$i
-		write_result_to_file "$TEST_NAME" "$TEST_ARGS" "$test_data" "$TEST_RESULT_FILE"
-	done
+
+	# Get PSS memory of CC components.
+	qemu_mem="$(get_pss_memory "$QEMU_BIN")"
+	cc_shim_mem="$(get_pss_memory "$SHIM_BIN")"
+	proxy_mem="$(get_pss_memory "$PROXY_BIN")"
+	cc_proxy_mem="$(echo "scale=2; $proxy_mem / $CC_NUMBER" | bc -l)"
+	cc_mem_usage=$(echo "scale=2; $qemu_mem + $cc_shim_mem + $cc_proxy_mem" | bc -l)
+
+	write_result_to_file "$test_name" "$test_args" "$cc_mem_usage" "$test_result_file"
```

Comment:
Now it should work, however in order to running yo need to execute:
```
make check
```
in order to expand the vars and create the scripts, and after you can execute it, or if you would like, you can execute all campaign running "run_docker_metrics" script in metrics dir.